### ISSUE TITLE
Migrate to Use New File Structure

### ISSUE THREAD

**Issue by emilyrohrbough** [cerner/terra-core#1257]
# Issue Description
With the deprecation of terra-core-site, all page and test examples should live together. The new file structure decided is as follows:
```
component
    /docs
    /examples
        /test-examples  # what use to be in the nightwatch dir
            test-name.example.jsx
        Index.site-page.jsx
    /src
    /tests
```

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [x] Other

## Expected Behavior
This update can roll out in increments. Once each package is updated to contain their own files, terra-clinical-site can be removed.

### Packages to be uplifted:
- [ ] alert
- [ ] arrange
- [ ] badge
- [ ] base
- [ ] button
- [ ] button-group
- [ ] card
- [ ] content-container
- [ ] date-picker
- [ ] demographics-banner
- [ ] dynamic-grid
- [ ] embedded-content-consumer
- [ ] form
- [ ] grid
- [ ] heading
- [ ] i18n
- [ ] icon
- [ ] image
- [ ] list
- [ ] markdown
- [ ] menu
- [ ] modal
- [ ] overlay
- [ ] popup
- [ ] progress-bar
- [ ] props-table
- [ ] responsive-element
- [ ] search-field
- [ ] slide-group
- [ ] slide-panel
- [ ] status
- [ ] status-view
- [ ] table
- [ ] text
- [ ] time-input
- [ ] toggle
- [ ] toggle-button

**Comment by emilyrohrbough** [comment]
We should split this migration into increments:
- Migrate all packages with wdio tests to use this file structure at one time
- For each package that needs to uplift its nightwatch tests to wdio, we should also update the examples to use this structure. This would remove all reference of nightwatch within these pages given the test examples currently live in the `package/tests/nightwatch` directory.

**Comment by emilyrohrbough** [comment]
Resolved in #1542
